WDL/Tree.py

```diff
@@ -1255,54 +1257,67 @@ def typecheck(self, check_quant: bool = True) -> None:
             self.workflow.typecheck(self, check_quant=check_quant)
 
 
-def load(
+async def read_source_default(uri: str, path: List[str], importer_uri: Optional[str]) -> str:
+    if uri.startswith("file://"):
+        uri = uri[7:]
+    if importer_uri:
+        if importer_uri.startswith("file://"):
+            importer_uri = importer_uri[7:]
+        importer_dir = os.path.dirname(importer_uri)
+        if importer_dir:
+            if not os.path.isabs(importer_dir):
+                morepaths = [os.path.join(p, importer_dir) for p in path]
+                path.extend(morepaths)
+            path.append(importer_dir)
```

Comment:
@dinvlad I want to highlight the above logic related to handling nested relative imports, which might become tricky for the LSP `get_document()` use case. For example suppose we have a directory tree,

```
A
├── a.wdl
└── B
    ├── b.wdl
    └── C
        └── c.wdl
```

and A/a.wdl does ``import B/b.wdl``, b.wdl does ``import C/c.wdl``. When asked to `read_source("C/C.wdl")` one has to figure out that's relative to `B/` not `A/`.

Reply:
Hmm, so the `uri` parameter to `read_source` is verbatim from `import`, not a fully resolved absolute path? I thought we could just the `uri` to LSP if it was a full `file://A/B/C/c.wdl`.

But if it's relative, we'll have to essentially use everything from `read_source_default()` until `with open(fn, "r") as infile:`, to fully resolve the `uri` but not to read it (as reading will be one from memory in LSP, or from file system if the `uri` is not in memory).

So given that we have to essentially reuse all but the last couple lines in `read_source_default()`, would it perhaps make sense to move the "full URI resolution" functionality into a separate function, and then to provide a `read_source_uri(uri)` method to actually implement loading from a fully-resolved URI?

Reply:
And for a fully-resolved network URI, we don't have to resolve it so it will be just short-cutted by  `read_source_default()`. So making the code in `read_source_default()` run always first, before calling `read_source_uri()`, could be a general solution :)